stream.go

```diff
@@ -102,21 +201,33 @@ func (s *StreamService) Dial(ctx context.Context, params *DialStreamParams) (*Co
 	}
 
 	conn := &Conn{
-		conn: websocketConn,
-		Msg:  make(chan StreamMessage),
+		conn:      websocketConn,
+		msg:       make(chan StreamMessage),
+		err:       make(chan error),
+		state:     StateConnected,
+		stateLock: &sync.Mutex{},
 	}
 
 	go func() {
 		defer conn.Close()
 		for {
 			var msg StreamMessage
 			if err := conn.conn.ReadJSON(&msg); err != nil {
+				if e, ok := err.(*websocket.CloseError); ok {
+					if isRevError, revError := IsRevError(e.Code); isRevError {
+						conn.err <- revError
+						return
+					}
+				}
+
 				if websocket.IsUnexpectedCloseError(err, websocket.CloseGoingAway, websocket.CloseAbnormalClosure) {
+					// perhaps an error should be sent on Err here too
+					conn.err <- err
 					return
 				}
 				continue
 			}
-			conn.Msg <- msg
+			conn.msg <- msg
```

Comment:
this channel is unbuffered so if noone is listening on this channel the msg will be missed, we may need to live with this though